extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -183,7 +184,7 @@ protected synchronized TaskStatus join(long timeout) throws IllegalStateExceptio
         since Druid doesn't support retrying tasks from a external system (K8s). We can explore adding a fabric8 watcher
         if we decide we need to change this later.
       **/
-      taskLocation = getTaskLocationFromK8s();
+      taskLocation.set(getTaskLocationFromK8s());
       updateState(new State[]{State.NOT_STARTED, State.PENDING}, State.RUNNING);
```

Comment:
i think we should reverse the order of these two lines. technically if updateState -> RUNNING is not called the runner will still not know where the task is even if the getTaskLocationAsync future returns (b/c getTaskLocation does a check on state first). in any case we should always be marking a task as running immediately after we join it (i looked through all the references to state)